spring-kafka/src/main/java/org/springframework/kafka/support/KafkaHeaders.java

```diff
@@ -67,4 +68,14 @@
 	 */
 	public static final String RECEIVED_PARTITION_ID = PREFIX + "receivedPartitionId";
 
+	/**
+	 * The header for holding the {@link org.apache.kafka.common.record.TimestampType type} of timestamp.
+	 */
+	public static final String TIMESTAMP_TYPE = PREFIX + "timestampType";
+
+	/**
+	 * The header for holding the timestamp of the record.
+	 */
+	public static final String TIMESTAMP = PREFIX + "timestamp";
```

Comment:
I think the clash `receive-and-send` is applied for `timestamp` as well.
So, we need one more header on the matter - `RECEIVED_TIMESTAMP`.
The same we have for `PARTITION_ID` and `TOPIC` and `KEY`.
I'm afraid you will have to redo converters logic appropriately.